### PR TITLE
(BEDS-155) Data Access: adjusted address struct

### PR DESCRIPTION
**Pull request by remoterami**
Returning `is_contract` and optional `label` in all cases where an address is retrieved.

### CODE REVIEW

File: backend/pkg/api/data_access/vdb_blocks.go [hunk 1]
```diff
@@ -382,7 +384,13 @@ func (d *DataAccessService) GetValidatorDashboardBlocks(ctx context.Context, das
 				Hash: t.Hash(hexutil.Encode(proposal.FeeRecipient)),
 			}
 			data[i].RewardRecipient = &rewardRecp
-			ensMapping[hexutil.Encode(proposal.FeeRecipient)] = ""
+			addressMapping[hexutil.Encode(proposal.FeeRecipient)] = nil
+			contractStatusRequests = append(contractStatusRequests, db.ContractInteractionAtRequest{
+				Address:  string(proposal.FeeRecipient),
```
Comment (Eisei24) [inline backend/pkg/api/data_access/vdb_blocks.go:389]: **issue:** I don't see this giving you the correct string formatting for `GetAddressContractInteractionsAt`.
It seems to request the address strings in the `fmt.Sprintf("%x", byteSlice)` format (so without the "0x") not as `string(byteSlice)`.
This affects all new uses of `GetAddressContractInteractionsAt` but please also see my comment in withdrawals.

File: backend/pkg/api/data_access/vdb_withdrawals.go [hunk 2]
```diff
@@ -393,12 +408,28 @@ func (d *DataAccessService) getNextWithdrawalRow(queryValidators []t.VDBValidato
 		withdrawalAmount = 0
 	}
 
+	ens_name, err := db.GetEnsNameForAddress(*address, utils.SlotToTime(nextWithdrawalSlot))
+	if err != sql.ErrNoRows {
+		return nil, err
+	}
+
+	contractStatusReq := []db.ContractInteractionAtRequest{{
+		Address: address.String(),
```
Comment (Eisei24) [inline backend/pkg/api/data_access/vdb_withdrawals.go:417]: **question:** I am not sure whether `address.String()` returns the string with the "0x" prefix, please check.

File: backend/pkg/commons/db/migrations/postgres/20240822134034_add_address_tags.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
+-- +goose Up
+-- +goose StatementBegin
+SELECT('up SQL query - create address_tags table');
+CREATE TABLE address_tags (
```
Comment (Eisei24) [inline backend/pkg/commons/db/migrations/postgres/20240822134034_add_address_tags.sql:4]: **minor:** Afaik should include `IF NOT EXISTS`.

File: backend/pkg/commons/db/migrations/postgres/20240822134034_add_address_tags.sql [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    address bytea NOT NULL UNIQUE,
+    tag CHARACTER VARYING(100) NOT NULL,
+    PRIMARY KEY (address, tag)
+);
+-- +goose StatementEnd
+
+-- +goose Down
+-- +goose StatementBegin
+SELECT('down SQL query - drop address_tags table');
+DROP TABLE address_tags;
```
Comment (Eisei24) [inline backend/pkg/commons/db/migrations/postgres/20240822134034_add_address_tags.sql:14]: **minor:** Afaik should include `IF EXISTS`.

**Review by Eisei24 — CHANGES_REQUESTED** [review]
Please check the address string conversion.

File: backend/pkg/api/data_access/vdb_withdrawals.go (hunk 2 above)
Comment (remoterami) [reply backend/pkg/api/data_access/vdb_withdrawals.go:417]: It does, also it's the checksum hashed address which would not match

File: backend/pkg/api/data_access/vdb_blocks.go (hunk 1 above)
Comment (remoterami) [reply backend/pkg/api/data_access/vdb_blocks.go:389]: Ah you're right, good, ty

**Review by Eisei24 — APPROVED** [review]
utACK

File: backend/pkg/commons/db/migrations/postgres/20240822134034_add_address_tags.sql [hunk 5]
```diff
@@ -0,0 +1,15 @@
+-- +goose Up
+-- +goose StatementBegin
+SELECT('up SQL query - create address_tags table');
+CREATE TABLE IF NOT EXISTS address_tags (
+    address bytea NOT NULL UNIQUE,
```
Comment (guybrush) [inline backend/pkg/commons/db/migrations/postgres/20240822134034_add_address_tags.sql:5]: maybe make this non-unique to able to have multiple tags per address?

**Review by guybrush — APPROVED** [review]

**Pull request merged**